src/sql/src/kafka_util.rs

```diff
@@ -237,6 +240,13 @@ pub async fn create_consumer(
     })
     .await
     .map_err(|e| sql_err!("{}", e))?;
+
+    // Check if the `DynamicBrokerRewritingClientContext` has
+    // errored first.
+    let error = context.dynamic_error();
+    if let Some(error) = error {
+        sql_bail!("setting up ssh tunnel: {}", error)
+    }
```

Comment:
There is an abstraction leak here, in that the method is called `dynamic_error` but the error message knows that the error is an SSH tunnel error. Propose we rename the method to `tunnel_error` and the change the error message to "setting up tunnel"; that seems to be nicely specific without breaking the abstraction boundary.